source/_posts/2021-06-21-job-hunting.markdown

```diff
@@ -0,0 +1,69 @@
+
+#### What to Prep
+* **Resume** --- update your resume with all the cool stuff you've been doing lately. Write down all of the interesting projects you've done at this job; you'll only end up putting a few on your resume, but having a list of things you've worked on lately will be helpful for preparing answers to behavioral questions ("tell me about a time a project ran long" etc.).
+* **System Design** --- this is a big one the more senior you get. You should be able to talk in broad strokes about how you'd design various systems, and be familiar (at least in passing) with database operations, caching, load balancing, etc. Some good resources:
+	* [Awesome System Design](https://github.com/madd86/awesome-system-design)
+	* [System Design Primer](https://github.com/donnemartin/system-design-primer)
+	* [System Design for Interviews](https://www.freecodecamp.org/news/systems-design-for-interviews/)
+	* [Grokking the System Design Interview](https://www.educative.io/courses/grokking-the-system-design-interview)
+    * Any number of system design/tech podcasts (I particularly liked using podcasts because it was interview prep I could do while out on a walk, instead of being glued to my computer)
+* **Coding & Algorithms** --- I was pleasantly surprised that few of my companies asked me algorithm-heavy "invert this binary tree"-type problems, but it's good to be prepared. Probably more than algorithms in particular though, for senior roles you should focus on being able to code cleanly, narrating your thought process as you go, and being able to talk about the time and space complexity of your solution. Additional things that will help you in coding interviews: liberal use of TODOs so you don't bog yourself down (e.g. "there's probably an edge case here, but let me finish the rest of the function and I'll come back to it"), and willingness to ask questions/use Google/admit when you don't know stuff ("ugh, is the function `.sort()` or `.sorted()`? Well, that's the sort of thing I'd Google, let's assume it's `.sort()` for now...").
+    * [interviews.school](https://interviews.school/)
+    * [Problem Solving with Algorithms and Data Structures using Python](https://runestone.academy/runestone/books/published/pythonds/index.html)
+    * [Leetcode](https://leetcode.com/)
+    * [Hackerrank](https://www.hackerrank.com/dashboard)
+    * Any ["Advent of Code"](https://adventofcode.com/) problems
+* **Behavioral Questions (aka "Soft Skills")** --- don't neglect this prep! Take the time to outline answers to the common behavioral questions ("tell me about a conflict you had with a coworker" etc.) and practice talking through them, so you have examples at your fingertips. In particular, it's useful to workshop these answers with a friend/collegue/mentor so that you can cast difficult situations in the best light, make sure you're being honest without being overly critical, etc.
```

Comment:
One of the ways I prepped for this was going through my own company's list of behavioral questions and making sure that I had answers to all of them